wrapper/src/main/java/software/amazon/jdbc/util/SlidingExpirationCache.java

```diff
@@ -118,9 +120,22 @@ protected void removeAndDispose(K key) {
   }
 
   protected void removeIfExpired(K key) {
-    final CacheItem cacheItem = cache.get(key);
-    if (cacheItem == null || cacheItem.shouldCleanup()) {
-      removeAndDispose(key);
+    // A list is used to store the cached item for later disposal since lambdas require references to outer variables
+    // to be final. This allows us to dispose of the item after it has been removed and the cache has been unlocked,
+    // which is important because the disposal function may be long-running.
+    final List<V> itemList = new ArrayList<>(1);
```

Comment:
Why do you use List<> instead of a simple variable of type V?

Reply:
If I try to do that I get "Variable used in lambda expression should be final or effectively final". The solutions my IDE proposes are other making it an atomic reference or making it a list. This relates to [this discussion](https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1122#discussion_r1767714929) on the custom endpoint PR.